ambry-messageformat/src/test/java/com.github.ambry.messageformat/MessageFormatInputStreamTest.java

```diff
@@ -443,4 +443,57 @@ private static void checkTtlUpdateSubRecord(InputStream stream, short accountId,
         updateRecord.getTtlUpdateSubRecord().getUpdatedExpiryTimeMs());
     Assert.assertEquals("UpdateTime mismatch", updateTimeMs, updateRecord.getUpdateTimeInMs());
   }
+
+  /**
+   * Test for {@link UndeleteMessageFormatInputStream}.
+   * @throws Exception
+   */
+  @Test
+  public void messageFormatUndeleteUpdateRecordTest() throws Exception {
+    StoreKey key = new MockId("id1");
+    short accountId = Utils.getRandomShort(TestUtils.RANDOM);
+    short containerId = Utils.getRandomShort(TestUtils.RANDOM);
+    long updateTimeMs = SystemTime.getInstance().milliseconds() + TestUtils.RANDOM.nextInt();
+    MessageFormatInputStream messageFormatInputStream =
+        new UndeleteMessageFormatInputStream(key, accountId, containerId, updateTimeMs);
+    long undeleteRecordSize = MessageFormatRecord.Update_Format_V3.getRecordSize(SubRecord.Type.UNDELETE);
+    int headerSize = MessageFormatRecord.getHeaderSizeForVersion(MessageFormatRecord.headerVersionToUse);
+    Assert.assertEquals(headerSize + undeleteRecordSize + key.sizeInBytes(), messageFormatInputStream.getSize());
+    checkUndeleteMessage(messageFormatInputStream, undeleteRecordSize, key, accountId, containerId, updateTimeMs);
+  }
+
+  /**
+   * Checks a undelete message including headers and the {@link UpdateRecord}.
+   * @param stream the {@link InputStream} to read data from
+   * @param expectedRecordSize the expected size of the record in the message. Can be {@code null} if unknown (won't be
+   *                            checked)
+   * @param key the expected {@link StoreKey}
+   * @param accountId the account id expected
+   * @param containerId the container id expected
+   * @param updateTimeMs the expected time of update
+   * @throws Exception any error.
+   */
+  private static void checkUndeleteMessage(InputStream stream, Long expectedRecordSize, StoreKey key, short accountId,
+      short containerId, long updateTimeMs) throws Exception {
+    checkHeaderAndStoreKeyForUpdate(stream, expectedRecordSize, key);
```

Comment:
if we decide to use V3 header in the future, please add lifeVersion check in `checkHeaderAndStoreKeyForUpdate` method.